### PR TITLE
Action step empty field ==> null

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

Now in the toolbar:

image

... and in the action editor:

image

... if you leave a field empty (`""`), it'll be saved as a `null` in the database.

The previous behaviour of saving `""` as the "Element text" was causing queries to fail, and couldn't be unticked in any way through the interface (no way in the app, bug in the toolbar). This together with https://github.com/PostHog/posthog/pull/5941 made it feel like there is something wrong with the action calculation.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

**Comment by timgl** [comment]
Assume this doesn't retroactively apply to existing actions. Should we add a fix in the backend as well?

File: frontend/src/scenes/actions/ActionStep.tsx [hunk 1]
```diff
@@ -124,7 +124,7 @@ function Option(props: {
     const onOptionChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void =>
         props.sendStep({
             ...props.step,
-            [props.item]: e.target.value,
+            [props.item]: e.target.value === '' ? null : e.target.value, // "" is a valid filter, we don't want it
```
Comment (alexkim205) [inline frontend/src/scenes/actions/ActionStep.tsx:127]: can this be simplified to `e.target.value || null` to catch all falsy values?

**Review by alexkim205 — APPROVED** [review]
left an inline comment should you choose to change it, but lgtm otherwise

File: frontend/src/scenes/actions/ActionStep.tsx (hunk 1 above)
Comment (mariusandra) [reply frontend/src/scenes/actions/ActionStep.tsx]: Definitely! :)

**Comment by mariusandra** [comment]
I added a migration to update existing actionsteps.

**Review by paolodamico — APPROVED** [review]
Have nothing to add other than what has already been said

**Review by alexkim205 — APPROVED** [review]

**Comment by Twixes** [comment]
This is good. I also added a query-time fix for this in https://github.com/PostHog/posthog/pull/6038. It shouldn't be truly needed if things work correctly from now on, but you never know, and besides we should have things function the same way across Postgres, ClickHouse, and the plugin server.

**Pull request merged**